pkg/reconciler/managed/reconciler.go

```diff
@@ -60,7 +61,7 @@ const (
 
 // Event reasons.
 const (
-	reasonCannotConnect       event.Reason = "CannotConnectToProvider"
+	reasonCannotDisconnect    event.Reason = "CannotDisconnectToProvider"
```

Comment:
```suggestion
	reasonCannotDisconnect    event.Reason = "CannotDisconnectFromProvider"
```

Also, I think `reasonCannotConnect` is still needed.